src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -160,6 +161,11 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
                 .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First());
 
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins;
+            // Emit at least one event per VS session, but not so many events
+            if (TransitiveDependenciesCounterfactual.ShouldEmitTelemetry)
```

Comment:
We should make sure emitting the telemetry actually succeeds and if not, reset the variable. If an exception is thrown inside the EmitTelemetryEvent method, the flag would be updated anyway.

Reply:
Added reset functionality in CounterfactualLogger.

Reply:
TelemetryOnceEmitter has this error handling funcionality.